spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/token/OAuth2AuthenticationConverter.java

```diff
@@ -0,0 +1,36 @@
+package com.sap.cloud.security.xsuaa.token;
+
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.provider.AuthorizationRequest;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+
+import com.sap.cloud.security.xsuaa.extractor.AuthoritiesExtractor;
+
+/**
+ * An authentication converter that sets a OAuth2Authentication object.
+ * Required to keep compatibility with UAA.
+ */
+public class OAuth2AuthenticationConverter extends AbstractAuthenticationConverter {
+
+
+	public OAuth2AuthenticationConverter(AuthoritiesExtractor authoritiesExtractor) {
+		super(authoritiesExtractor);
+	}
+
+	@Override
+	public OAuth2Authentication convert(Jwt jwt) {
+		XsuaaToken token = new XsuaaToken(jwt);
+		AuthorizationRequest authorizationRequest = new AuthorizationRequest(token.getClientId(),
+				authoritiesExtractor.getAuthorities(token).stream()
+						.map(object -> Objects.toString(object, null))
+						.collect(Collectors.toList()));
+		authorizationRequest.setApproved(true);
+		authorizationRequest.setAuthorities(authoritiesExtractor.getAuthorities(token));
```

Comment:
why do you need to add scopes as String Collection as part of the constructor AND as GrantedAuthority as part of setAuthorities method? Preferred... pass them via method and keep constructor null, if possible

Reply:
will try if that works